src/main/scala/cognite/spark/v1/SequencesRelation.scala

```diff
@@ -141,25 +131,19 @@ class SequencesRelation(config: RelationConfig)(val sqlContext: SQLContext)
       .flatTap(_ => incMetrics(itemsDeleted, ids.length))
   }
 
-  override def upsert(rows: Seq[Row]): IO[Unit] = {
-    val sequences =
-      rows
-        .map(fromRow[SequenceUpsertSchema](_))
-
-    implicit val toCreate =
-      Transformer
-        .define[SequenceUpsertSchema, SequenceCreate]
-        .withFieldComputed(
-          _.columns,
-          x => x.getSequenceColumnCreate
-        )
-        .buildTransformer
-
-    val existingSeqs: Map[Option[Long], Seq[String]] = getExistingSequenceColumnsByIds(sequences)
-    val existingSeqsWithExtId: Map[Option[String], Seq[String]] =
-      getExistingSequenceColumnsByExternalIds(sequences)
+  private def transformerUpsertToCreate() =
```

Comment:
this didn't work for me when I tried, maybe it was sth else that I fixed :D :D thanks @vuhainguyen, probably the implicit parameters that I added to some helpers were missing and I blamed this instead...

Reply:
👯 :D

Reply:
may be you can move the bloc below 
```scala
    val existingSeqs: Map[Option[Long], Seq[String]] = getExistingSequenceColumnsByIds(sequences)
    val existingSeqsWithExtId: Map[Option[String], Seq[String]] =
      getExistingSequenceColumnsByExternalIds(sequences)
```
into the transformer update to avoid duplicate code as well 😁

Reply:
done  :p